code/modules/fishing/sources/source_types.dm

```diff
@@ -773,3 +774,165 @@
+	///Using more than the money needed to buy the most expensive item (why would you do it?!) will remove the dud chance.
+	if(bait_value > highest_price)
+		table -= FISHING_DUD
+	else
+		//Makes using 1 cred chips with the minigame skip (negative fishing difficulty) a bit less cheesy.
+		var/malus = min(PAYCHECK_LOWER - bait_value, highest_price)
+		if(malus > 0)
+			table[FISHING_DUD] += malus * malus_multiplier
+			table[FISHING_VENDING_CHUCK] += malus * malus_multiplier
+
+#define FISHING_PRODUCT_DIFFICULTY_MULT 1.6
+
+/datum/fish_source/vending/calculate_difficulty(result, obj/item/fishing_rod/rod, mob/fisherman)
+	//Using less than a minimum paycheck is going to make the challenge a tad harder.
+	var/bait_value = rod.bait?.get_item_credit_value()
+	var/base_diff = PAYCHECK_LOWER - bait_value
+	return ..() + get_product_difficulty(base_diff, result) * FISHING_PRODUCT_DIFFICULTY_MULT
+
+/datum/fish_source/vending/proc/get_product_difficulty(diff, datum/result)
+	if(istype(result, /datum/data/vending_product))
+		var/datum/data/vending_product/product = result
+		diff = min(diff, product.price) // low priced items are easier to catch anyway
+	return diff
+
+#undef FISHING_PRODUCT_DIFFICULTY_MULT
+
+/datum/fish_source/vending/dispense_reward(reward_path, mob/fisherman, atom/fishing_spot)
+	if(reward_path != FISHING_VENDING_CHUCK)
+		return ..()
+	var/obj/machinery/vending/vending = fishing_spot
+	if(istype(fishing_spot, /obj/machinery/fishing_portal_generator))
+		var/obj/machinery/fishing_portal_generator/portal = fishing_spot
+		vending = portal.current_linked_atom
+	if(fishing_spot != vending) //fishing portals
+		vending.forceMove(get_turf(fishing_spot))
+	vending.tilt(fisherman, range = 4)
+	return null //Don't spawn a reward at all
+
+/datum/fish_source/vending/spawn_reward(reward_path, atom/spawn_location, obj/machinery/vending/fishing_spot)
+	if(istype(fishing_spot, /obj/machinery/fishing_portal_generator))
+		var/obj/machinery/fishing_portal_generator/portal = fishing_spot
+		fishing_spot = portal.current_linked_atom
+	if(!istype(fishing_spot))
+		return null
+	return spawn_vending_reward(reward_path, spawn_location, fishing_spot)
+
+/datum/fish_source/vending/proc/spawn_vending_reward(reward_path, atom/spawn_location, obj/machinery/vending/fishing_spot)
+	var/datum/data/vending_product/product_record = reward_path
+	if(!istype(product_record) || product_record.amount <= 0)
+		return null
+	return fishing_spot.dispense(product_record, spawn_location)
+
+/datum/fish_source/vending/pre_challenge_started(obj/item/fishing_rod/rod, mob/user, datum/fishing_challenge/challenge)
+	RegisterSignal(rod, COMSIG_FISHING_ROD_CAUGHT_FISH, PROC_REF(on_reward))
+
+/datum/fish_source/vending/proc/on_reward(obj/item/fishing_rod/rod, atom/movable/reward, mob/user)
+	SIGNAL_HANDLER
+	if(reward && !QDELETED(rod.bait) && rod.bait.get_item_credit_value()) //you pay for what you get
+		qdel(rod.bait) // fishing_rod.Exited() will handle clearing the hard ref.
```

Comment:
I don't really have a better spot to add this comment, but for consistency with the vending machine  restock gameplay-loop, the vending machine's credits_contained should add however many credits the item originally cost. I'm avoiding just adding the number of bait-credits directly to avoid cargo players just reusing like, 100 credits to empty out a vending machine for free due to their increased access to vending machine restocks.